### PR TITLE
Plasma statement example

### PR DESCRIPTION
**Pull request by syuhei176**
- [x] Add IncludedInIntervalTreeAtBlockDecider
- [x] Add BlockRangeQuantifier
- [ ] Property and Witness en/decoding

### CODE REVIEW

**Comment by syuhei176** [comment]
IncludedInIntervalTreeAtBlockDecider was finished. I'm breaking down into small PRs with refactoring. see #156, #157, and coming soon other PRs.

File: ovm/src/lib.rs [hunk 1]
```diff
@@ -143,13 +144,50 @@ mod tests {
...
+                    panic!("invalid type in PropertyFactory");
+                }
+            })),
+            None,
+        )))
+    }
+
+    /// plasma
+    #[test]
+    fn test_fail_to_decide_plasma_checkpoint() {
```
Comment (syuhei176) [inline ovm/src/lib.rs]: Added plasma checkpoint property example.

**Comment by syuhei176** [comment]
Make ABI en/decoding another task.

File: ovm/src/types/witness.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            ],
+            data,
+        )
+        .map_err(|_e| PlasmaCoreError::from(PlasmaCoreErrorKind::AbiDecode))?;
+        Self::from_tuple(&decoded)
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum Witness {
```
Comment (tkmct) [inline ovm/src/types/witness.rs]: This is great!

**Pull request merged**